File: common/src/main/java/dev/pandasystems/pandalib/client/PandaLibClient.java
/*
 * Copyright (C) 2025 Oliver Froberg (The Panda Oliver)
 *
 * This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  any later version.
 *
 * You should have received a copy of the GNU Lesser General Public License
 *  along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package dev.pandasystems.pandalib.client;

import dev.pandasystems.pandalib.client.resource.AssimpResources;
import dev.pandasystems.pandalib.core.client.event.EventHandlerClient;
import dev.pandasystems.pandalib.registry.ReloadListenerRegistry;
import net.minecraft.server.packs.PackType;

public class PandaLibClient {
	private static PandaLibClient instance;
	
	public final AssimpResources assimpResources = new AssimpResources();
	
    public PandaLibClient() {
		instance = this;
		EventHandlerClient.init();

		ReloadListenerRegistry.register(PackType.CLIENT_RESOURCES, assimpResources);
    }

	public static PandaLibClient getInstance() {
		return instance;
	}
}

File: common/src/main/kotlin/client/PandaLibClient.kt
/*
 * Copyright (c) 2025. Oliver Froberg
 *
 * This code is licensed under the GNU Lesser General Public License v3.0
 * See: https://www.gnu.org/licenses/lgpl-3.0-standalone.html
 */

package dev.pandasystems.pandalib.client

import dev.pandasystems.pandalib.client.config.ClientConfigSynchronizer
import dev.pandasystems.pandalib.pandalibLogger
import dev.pandasystems.pandalib.utils.InternalPandaLibApi


@InternalPandaLibApi
fun initializePandaLibClient() {
	pandalibLogger.debug("PandaLib Client is initializing...")

	ClientConfigSynchronizer.init()

	pandalibLogger.debug("PandaLib initialized successfully.")
}